lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb

```diff
@@ -0,0 +1,107 @@
+require 'fastlane/action'
+require 'digest/sha1'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      S3_UPLOADED_FILE_PATH = :S3_UPLOADED_FILE_PATH
+    end
+
+    class UploadToS3Action < Action
+      def self.run(params)
+        file_path = params[:file]
+        file_name = File.basename(file_path)
+        UI.user_error!("Unable to read file at #{file_path}") unless File.file?(file_path)
+
+        bucket = params[:bucket]
+        key = params[:key]
+
+        if params[:auto_prefix] == true
+          file_name_hash = Digest::SHA1.base64digest(file_name)
+          key = [file_name_hash, key].join('/')
+        end
+
+        UI.user_error!("File already exists at #{key}") if file_is_already_uploaded?(bucket, key)
+
+        UI.message("Uploading #{file_path} to: #{key}")
+
+        File.open(file_path, 'rb') do |file|
+          Aws::S3::Client.new().put_object({
+                                             body: file,
+                                             bucket: bucket,
+                                             key: key
+                                           })
+        rescue Aws::S3::Errors::ServiceError => e
```

Comment:
Would it be worth rescuing other errors, too?

Reply:
☝️

Reply:
I thought about that, but my thinking was that most common issue will be "S3 is having issues or there's internet weather or AWS is down _again_", and if something _else_ is wrong, I'd rather rather see a stack trace in order to figure out exactly where the problem happened (vs a simple exception `message` that might not be helpful). 🤷‍♂️